clang/include/clang/Sema/SemaHLSL.h

```diff
@@ -30,9 +31,9 @@ class Scope;
 
 // FIXME: This can be hidden (as static function in SemaHLSL.cpp) once we no
 // longer need to create builtin buffer types in HLSLExternalSemaSource.
-bool CreateHLSLAttributedResourceType(Sema &S, QualType Wrapped,
-                                      ArrayRef<const Attr *> AttrList,
-                                      QualType &ResType);
+bool CreateHLSLAttributedResourceType(
+    Sema &S, QualType Wrapped, ArrayRef<const Attr *> AttrList,
+    QualType &ResType, HLSLAttributedResourceLocInfo *LocInfo = nullptr);
```

Comment:
Do we actually need the `HLSLAttributedResourceLocInfo` to be defaulted/nullable? Can it just be a reference instead?

Reply:
We do not have typeloc info to fill out when we are creating implicit buffer types in HLSLExternalSemaSource, so the nullptr means do not fill.